lib/xml/parser.js

```diff
@@ -1,23 +1,80 @@
 "use strict";
 
 const { promisify } = require("util");
-const { parseString } = require("xml2js");
-const parseXMLString = promisify(parseString);
+const xml2js = require("xml2js");
 const ICAL = require("ical.js");
 const moment = require("moment");
 
+// setup XML parse so that it always generates a $ns for the elements and also strip the prefixes
+// from the names. some servers (e.g. iCloud) use an empty prefix for the DAV namespace.
+const parser = new xml2js.Parser({
+    tagNameProcessors: [
+        xml2js.processors.stripPrefix
+    ],
+    explicitCharkey: true,
+    xmlns: true
+});
```

Comment:
We'd like to avoid **foldCharacterElements** as there's a lot of stuff happening in there. What do you think about not using `xmlns: true` and just stripping the namespace prefixes? This way we would not need to flatten the xml.

Reply:
Another idea: We could try to find out if the xml makes use of the `d:` namespace and have it as some sort of variable in our codebase.

```js
// First we'd need to find out if the xml uses a certain namespace
// Could be "d:" or "" or something else.
const ns = findTheNamespace();

// We could access the object's properties like so:
parsed.href = result[`${ns}multistatus`][`${ns}response`][0][`${ns}href`][0];

// Or maybe create some sort of abstraction for adding the namespace to a given string.
```

Reply:
> We'd like to avoid **foldCharacterElements** as there's a lot of stuff happening in there. What do you think about not using `xmlns: true` and just stripping the namespace prefixes? This way we would not need to flatten the xml and could still get rid of "d:" in the codebase.

the problem is, that the XML parser automatically enables `xmlns` when you enable the stripping of the prefixes.